flink-python/src/main/java/org/apache/flink/table/functions/python/coders/RowCoder.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.functions.python.coders;
+
+import org.apache.flink.annotation.Internal;
+import org.apache.flink.types.Row;
+import org.apache.flink.util.Preconditions;
+
+import org.apache.beam.sdk.coders.Coder;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * A {@link Coder} for {@link Row}.
+ */
+@Internal
+public class RowCoder extends Coder<Row> {
+
+	private static final long serialVersionUID = 1L;
+
+	private final Coder<Object>[] fieldCoders;
+
+	@SuppressWarnings("unchecked")
+	public RowCoder(Coder<?>[] fieldCoders) {
+		this.fieldCoders = (Coder<Object>[]) Preconditions.checkNotNull(fieldCoders);
+	}
+
+	public Coder<?>[] getFieldCoders() {
+		return this.fieldCoders;
+	}
+
+	@Override
+	public void encode(Row row, OutputStream outStream) throws IOException {
+		int len = fieldCoders.length;
+
+		if (row.getArity() != len) {
+			throw new RuntimeException("Row arity of input element does not match coders.");
+		}
+
+		for (int i = 0; i < row.getArity(); i++) {
+			fieldCoders[i].encode(row.getField(i), outStream);
```

Comment:
Should take the null field into consideration. Coder in Beam can not encode null values. Same for `BaseRowCoder`.